test/libyul/yulSyntaxTests/eof/eofcreate.yul

```diff
@@ -0,0 +1,13 @@
+object "a" {
+    code {
+        let addr := eofcreate("b", 0, 0, 0, 0)
```

Comment:
You might want to also cover the case where the name refers to `data` rather than an object.

Reply:
Yes. But this should generate an error. Do wanna fix this in https://github.com/ethereum/solidity/pull/15536 ?

Reply:
I implemented this in separated commit https://github.com/ethereum/solidity/pull/15536/commits/d0e7890c6e265b6ffb3143baafa42d9a520eeee2
Let me know if it's ok besides naming which has to be fixed.

Reply:
The test needs to be here, because in #15536 you don't have `eofcreate()` and `returncontract()` and those are the only things that you can test it with.

And we don't *have to* distinguish objects from data just for this test, it should already work on EOF because only object names are supposed to be accepted. The test is only meant to confirm that it works correctly. But yeah, being able to distiguish them would not hurt because it would let us generate a more specific error message rather than just saying "this object does not exist".

The general idea with the struct and separating data from objects is good.

Reply:
Added.